src/react/databrowser/objects/details/Metadata.jsx

```diff
@@ -112,8 +115,22 @@ function Properties({ objectMetadata }: Props) {
         if (!isValidItems) {
             return;
         }
-        const metadata = convertToAWSMetadata(items);
-        dispatch(putObjectMetadata(bucketName, prefixWithSlash, objectKey, metadata));
+        const { systemMetadata, userMetadata } = convertToAWSMetadata(items);
+        dispatch(putObjectMetadata(bucketName, prefixWithSlash, objectKey, systemMetadata, userMetadata));
+    };
+
+    const selectValue = (metadata: MetadataItem) => {
+        const { key, type } = metadata;
+        if (type) {
+            if (isSystemMetadata(type) && key) {
+                return selectOptions.find(l => l.value === key);
+            }
+            if (isUserMetadata(type)) {
+                return userMetadataOption;
+            }
```

Comment:
Based on my suggestion above for the isXMetadata function this could look like:
```suggestion
        if (isSystemMetadata(metadata) && metadata.key) {
            return selectOptions.find(l => l.value === key);
        }
        if (isUserMetadata(metadata)) {
            return userMetadataOption;
        }
```